### DISCUSSION TITLE
Adjust the duration of a spell's turn to an actual turn rather than the current turn

### DISCUSSION THREAD

**Discussion by LeHerosInconnu** [ihhub/fheroes2#6691] [Ideas]
I suggest to adjust the duration of a spell's turn to "an actual turn" rather than "the current turn".

In the example, the Cyclops troop was paralyzed for one turn.
This troop is the fastest of the troops on the battlefield and played before being paralyzed.
Since the spell only lasts for the current turn, the spell will no longer be active at the end of the current turn and the Cyclops troop will play first the next turn.
The spell was therefore cast for nothing.

In fheroes2:

![Problem paralyze current turn 01](https://user-images.githubusercontent.com/43583994/220611617-cdefd129-bc84-476f-aa51-f6e6e14629e8.png)

**Comment by oleg-derevenetz** [comment]
> Since the spell only lasts for the current turn, the spell will no longer be active at the end of the current turn and the Cyclops troop will play first the next turn.

So if you have a weak hero with low spell power, you should suffer and carefully choose the target for your spells. This is by design.

**Reply by LeHerosInconnu** [reply]
Hello @oleg-derevenetz,

> > Since the spell only lasts for the current turn, the spell will no longer be active at the end of the current turn and the Cyclops troop will play first the next turn.
> 
> So if you have a weak hero with low spell power, you should suffer and carefully choose the target for your spells. This is by design.

Even a hero with more spell power would be penalized, the duration of the spell is, in fact, reduced by one turn compared to what is displayed.

**Reply by oleg-derevenetz** [reply]
> the duration of the spell is, in fact, reduced by one turn compared to what is displayed

No if the target unit still didn't get its turn. That's why I mentioned that with a weak hero spell targets should be considered carefully :) You should either have more spell power or have a faster unit in your army and get the first turn.

**Reply by LeHerosInconnu** [reply]
@oleg-derevenetz.

> > the duration of the spell is, in fact, reduced by one turn compared to what is displayed
> 
> No if the target unit still didn't get its turn. That's why I mentioned that with a weak hero spell targets should be considered carefully :) You should either have more spell power or have a faster unit in your army and get the first turn.

The problem is that a spell can be cast when it will have no effect.
So why allow the spell to be cast at all?

**Reply by oleg-derevenetz** [reply]
> So why allow the spell to be cast at all? 

For the same reason that players are generally allowed to make mistakes, and there is no button "just win the game for me" :)

**Reply by Branikolog** [reply]
Hi, @LeHerosInconnu !

>The spell was therefore cast for nothing.

This is not strictly true. 
I sometimes use "paralyze" to be able to attack the target with my fragile troop without retaliation. :)
For the same purposes I sometimes use "blind" spell too.

**Comment by LeHerosInconnu** [comment]
Maybe I'll revisit that later.
I close the discussion.
